crates/sui-graphql-rpc/src/types/gas.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::types::object::Object;
+use async_graphql::*;
+
+use super::{address::Address, big_int::BigInt};
+
+#[derive(Clone, Debug, PartialEq, Eq, SimpleObject)]
+pub(crate) struct GasInput {
+    pub gas_sponsor: Option<Address>,
+    pub gas_payment: Option<Vec<Object>>,
+    pub gas_price: Option<BigInt>,
+    pub gas_budget: Option<BigInt>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, SimpleObject)]
+pub(crate) struct GasCostSummary {
+    pub computation_cost: Option<BigInt>,
```

Comment:
why "Option<>"here? GasCostSummary in sui-types does not have Option wrapper.

Reply:
It's for more flexibility within a given RPC version, to make changes -- e.g. it means that the server can start producing a `None` response for this if, for some reason, we no longer want to support it, or we want to rename it, etc.